tests/databases/molecule/mongo_db/test_molecule_update.py

```diff
@@ -0,0 +1,255 @@
+import stk
+import numpy as np
+import pymongo
+
+from .utilities import get_database_state
+from ...utilities import (
+    DatabaseEntry,
+    DatabaseState,
+    assert_database_state,
+)
+
+
+def to_hashable(json):
+    json = dict(json)
+    return {
+        'm': tuple(tuple(row) for row in json.pop('m')),
+        **json,
+    }
+
+
+def test_update_1():
+    """
+    Test that existing entries are updated.
+
+    """
+
+    database_name = '_test_update_1'
+    client = pymongo.MongoClient()
+    client.drop_database(database_name)
+
+    database = stk.MoleculeMongoDb(
+        mongo_client=client,
+        database=database_name,
+        put_lru_cache_size=0,
+        get_lru_cache_size=0,
+    )
+    jsonizer = stk.MoleculeJsonizer()
+
+    molecule = stk.BuildingBlock('CCC').with_canonical_atom_ordering()
+    json = jsonizer.to_json(molecule)
+
+    database.put(molecule)
+    assert_database_state(
+        state1=get_database_state(database),
+        state2=DatabaseState({
+            DatabaseEntry(**json['molecule']): 1,
+            DatabaseEntry(**to_hashable(json['matrix'])): 1,
+        }),
```

Comment:
So a single .put call gives you two entries? Or does state2 here collapse into 1 entry?